### DISCUSSION TITLE
Feedback Request: How are you currently using `updated_checkout` event

### DISCUSSION THREAD

**Discussion by senadir** [woocommerce/woocommerce#43773] [Checkout Flow]
As we're trying to find an equivalence for the `updated_checkout` event in the new Checkout Block, we want to understand if it make sense to have a single overarching event, or focus more on use cases.

This is because mainly, Checkout block is a client side app that as a lot of updates, some updates go to the server, some don't, and we would need to understand your use cases and expectations to solve this.

**Comment by MichaelBengtsson** [comment]
I am assuming you mean the JavaScript event `updated_checkout`, since i can't find a reference to `checkout_updated` anywhere when i made a quick search just now. But if i'm wrong on that, let me know!

We use both the `updated_checkout` event and the related `update_checkout` event.
For our payment methods we use the `update_checkout` event regularly when we have to make changes to the checkout. We use this to not have to make our own AJAX calls since that could cause conflicts with the WooCommerce sessions if other AJAX calls happened at the same time, and they also try to write to the session.

Our embedded  checkouts, for example [Klarna Checkout](https://wordpress.org/plugins/klarna-checkout-for-woocommerce/), [Nets Easy](https://wordpress.org/plugins/dibs-easy-for-woocommerce/), [Payson Checkout](https://wordpress.org/plugins/krokedil-paysoncheckout-20-for-woocommerce/), have their own input fields for address data. When this data is sent to us by their respective JavaScript APIs, we fill the form fields in WooCommerce and trigger the `update_checkout` event to have WooCommerce calculate new shipping, and update the data to the session.
Some of our embedded checkouts will also have integrated shipment selection as well. When this change is done, we also trigger the `update_checkout` event to trigger a recalculation of the WooCommerce cart, and to ensure that other plugins can also use the updated shipping method and costs.

We also listen to both the `update_checkout` and the `updated_checkout` events to suspend and resume the embedded checkouts. To prevent customers from inputting data during the time calculations are running in WooCommerce that could be triggered by other plugins.

**Reply by senadir** [reply]
> I am assuming you mean the JavaScript event updated_checkout, since i can't find a reference to checkout_updated anywhere when i made a quick search just now. But if i'm wrong on that, let me know!

Thanks! I updated the discussion body!
